src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -270,7 +270,18 @@ foam.CLASS({
         nu = old.fclone();
         nu.copyFrom(obj);
       }
-      return dao.put(nu);
+
+      try {
+        return dao.put(nu);
+      } catch ( ValidationException ve ) {
+        throw new DAOPutException(ve.getMessage());
```

Comment:
do you need to main the original exception?
```suggestion
        throw new DAOPutException(ve.getMessage(), ve);
```

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4772/commits/5d03767cfca7ad2e108716f8a94212e953cb0dca.